### ISSUE TITLE
Invalid unicode character in conversion of comment

### ISSUE THREAD

**Issue by mikemccand** [apache/lucene-jira-archive#60]
Spot checking a few converted issues, I noticed the invalid Unicode character, I think (U+FFDD) in [this comment](https://github.com/mikemccand/stargazers-migration-test/issues/329#issuecomment-1192052095) but the [corresponding Jira issue comment](https://issues.apache.org/jira/browse/LUCENE-8329?focusedCommentId=16487052&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16487052) seems to have just a whitespace character.

Not sure how widespread this issue is.

**Comment by mikemccand** [comment]
Sorry, I erased and regenerated my repo, breaking the above comment link.  It is now [here](https://github.com/mikemccand/stargazers-migration-test/issues/329#issuecomment-1193207785).

This is the original content from Jira:

```
          "body": "Patch is available but it involves a binary file change.\u0010\r\nSo no easy to review and not easy to check with different OS",
```

I'm not yet sure exactly how that `\u0010` is supposed to be interpreted -- is that hex or decimal in JSON strings?  I'll try to dig and see whether other Jira-exported issues use unicode escapes like this.

**Comment by mikemccand** [comment]
> I'm not yet sure exactly how that `\u0010` is supposed to be interpreted -

OK it looks like [it is hexadecimal](https://tools.ietf.org/id/draft-ietf-json-rfc4627bis-09.html#rfc.section.7), so `\u0010` is this [exotic control character](https://www.fileformat.info/info/unicode/char/0010/index.htm).  I'm not sure why it was not recognized, exotic though it be, on import, and converted to `U+FFFD` (the [fallback Unicode Replacement Character](https://www.fileformat.info/info/unicode/char/fffd/index.htm) when something goes wrong in decoding Unicode).

**Comment by mikemccand** [comment]
Indeed, I can see that we converted it "correctly" (preserving the U+0010 character):

```
      "body": "Patch is available but it involves a binary file change.\u0010\nSo no easy to review and not easy to check with different OS\n\n[Legacy Jira: Alessandro Benedetti (@alessandrobenedetti) on [May 23 2018](https://issues.apache.org/jira/browse/LUCENE-8329?focusedCommentId=16487052&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16487052)]\n",
```

Maybe GitHub's import API does not allow "exotic" (yet, correctly encoded) Unicode characters and remaps some of them to U+FFFD?  I tried searching for � LOL in GitHub's search but it matches nothing.  I guess they do not index that character as a separate token ;)

**Comment by mikemccand** [comment]
Let's see what happens if I try to add this character myself: 

Patch is available but it involves a binary file change.
So no easy to review and not easy to check with different OS

**Comment by mikemccand** [comment]
Whoa, indeed, even if you try to enter this character, GitHub remaps it.  I wonder what else they remap?

**Comment by mocobeta** [comment]
Regarding to "\u0010", fortunately it's relatively rare. 
```
migration $ find jira-dump -name "*.json" | xargs -I{} cat {} | grep "\u0010" | wc -l
13
```
Not sure about other ctrl characters.

**Comment by mikemccand** [comment]
> Regarding to "\u0010", fortunately it's relatively rare.

Phew, thanks for checking.  Kinda crazy this exotic character occurs that many time in our Jira issues!  Let's ignore it.

I was going to try to figure out how many such remapped characters are happening, so I am trying to run `remap_cross_issue_links.py`.  But first I discovered that in fact the character IS stored correctly in GitHub for this suspect issue comment!  (I.e. on export it is still U+000A).  So it must be in GH's front-end rendering of an issue where this U+FFFD substitution takes place.

**Comment by mikemccand** [comment]
Reminder: once we have the draft migrated test repo public, find some Jira issues that have exotic Unicode escapes/characters and confirm that migrated properly.
